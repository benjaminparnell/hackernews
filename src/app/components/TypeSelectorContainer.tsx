import * as React from 'react';
import styled from 'styled-components';
import { clearFix } from 'polished';
import { StoryType } from '../types';

type Props = {
  setActiveTab: (type: StoryType) => void;
  activeTab: StoryType;
};

const TypeButton = styled.button<{ active: boolean }>`
  border: 2px solid ${props => props.theme.primary};
  color: ${props =>
    props.active ? props.theme.background : props.theme.primary};
  font-size: 1.2em;
  padding: 0.75em 1.1em;
  width: 50%;
  background-color: ${props =>
    props.active ? props.theme.primary : 'transparent'};
`;

const ButtonWrapper = styled.div`
  ${clearFix()}
`;

const TYPES: { label: string; type: StoryType }[] = [
  { label: 'Top', type: 'top' },
  { label: 'New', type: 'new' }
];

const TypeSelector = ({ setActiveTab, activeTab }: Props) => (
  <ButtonWrapper>
    {TYPES.map(({ label, type }) => (
      <TypeButton
        key={type}
        type="button"
        onClick={() => setActiveTab(type)}
        active={activeTab === type}
      >
        {label}
      </TypeButton>
    ))}
  </ButtonWrapper>
);

export default TypeSelector;
