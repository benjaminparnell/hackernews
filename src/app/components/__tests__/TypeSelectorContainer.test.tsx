import * as React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import TypeSelectorContainer, { TYPES } from '../TypeSelectorContainer';

describe('<TypeSelectorContainer />', () => {
  afterEach(cleanup);

  it('should match the snapshot', () => {
    expect(
      render(<TypeSelectorContainer activeTab="new" setActiveTab={() => {}} />)
        .container.firstChild
    ).toMatchSnapshot();
  });

  it('should render buttons for all tabs in TYPES', () => {
    const { getByText } = render(
      <TypeSelectorContainer activeTab="new" setActiveTab={() => {}} />
    );
    TYPES.forEach(({ label }) => getByText(label));
  });

  it('should call the setActiveTab(type) when the button for a type is clicked', () => {
    const setActiveTabSpy = jest.fn();
    const { getByText } = render(
      <TypeSelectorContainer activeTab="new" setActiveTab={setActiveTabSpy} />
    );
    fireEvent.click(getByText('Top'));
    expect(setActiveTabSpy).toHaveBeenCalledTimes(1);
    expect(setActiveTabSpy).toHaveBeenCalledWith('top');
  });
});
