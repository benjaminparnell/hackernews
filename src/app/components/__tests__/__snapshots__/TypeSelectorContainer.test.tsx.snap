// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeSelectorContainer /> should match the snapshot 1`] = `
<div
  class="sc-bwzfXH eqBUmh"
>
  <button
    class="sc-bdVaJa bNGhWs"
    type="button"
  >
    Top
  </button>
  <button
    class="sc-bdVaJa fgRJQW"
    type="button"
  >
    New
  </button>
</div>
`;
